Assets/Scripts/Framework/Gameplay/Interactions/Grabbables/Grabbable.cs

```diff
@@ -44,25 +62,48 @@ private protected override bool Interact(Interactor interactor)
         [Server]
         private protected virtual void UseGrabbable(Interactor interactor)
         {
-            interactor.UpdateInteractingTarget(this, IsBeingInteracted);
-
-            if (IsBeingInteracted)
+            if (!IsBeingInteracted)
             {
-                NetworkObject.SetParent(interactor.GrabbingAnchor.GetComponent<NetworkBehaviour>());
+                NetworkObject.SetParent(interactor.GrabbingAnchor);
                 transform.localPosition = Vector3.zero;
+
+                UpdateInteractionStatus(interactor, true);
+                interactor.UpdateInteractingTarget(interactor.Owner, this, IsBeingInteracted);
+                return;
             }
-            else
-            {
-                NetworkObject.UnsetParent();
 
-                if (!gameObject.TryGetComponent(out BoxCollider collider))
-                    return;
+            NetworkObject.UnsetParent();
+            UpdateInteractionStatus(null, false);
 
-                Physics.Raycast(transform.position, Vector3.down, out RaycastHit hit, _groundCheckDistance);
+            interactor.UpdateInteractingTarget(interactor.Owner, null, IsBeingInteracted);
+            PlaceOnGround();
+            return;
+        }
 
-                hit.point += new Vector3(0, collider.size.y * .5f, 0);
-                transform.position = hit.point;
-            }
+        /// <summary>
+        /// A function that updates the interaction information of this <see cref="Grabbable"/>.
+        /// </summary>
+        /// <param name="interactor">The current <see cref="Interactor"/> of this <see cref="Grabbable"/>.</param>
+        /// <param name="isBeingInteracted">The new IsBeingInteracted bool state.</param>
+        public void UpdateInteractionStatus(Interactor interactor, bool isBeingInteracted)
+        {
+            IsBeingInteracted = isBeingInteracted;
+            _originInteractor = interactor;
+        }
+
+        /// <summary>
+        /// A function that places the <see cref="Grabbable"/> on the ground, can be used for after snapping to a
+        /// transform.
+        /// </summary>
+        public virtual void PlaceOnGround()
+        {
+            if (!gameObject.TryGetComponent(out BoxCollider collider))
+                return;
+
+            Physics.Raycast(transform.position, Vector3.down, out RaycastHit hit, _groundCheckDistance);
+
+            hit.point += new Vector3(0, collider.size.y * .5f, 0);
+            transform.position = hit.point;
```

Comment:
In my opinion this is better.
```suggestion
            transform.position = hit.point + new Vector3(0, collider.size.y * .5f, 0);
```